src/pool/srv_iv.c

```diff
@@ -814,24 +814,26 @@ pool_iv_ent_fetch(struct ds_iv_entry *entry, struct ds_iv_key *key,
 int
 ds_pool_iv_refresh_hdl(struct ds_pool *pool, struct pool_iv_hdl *pih)
 {
-	int rc;
+	int rc = 0;
 
-	if (!uuid_is_null(pool->sp_srv_cont_hdl)) {
-		if (uuid_compare(pool->sp_srv_cont_hdl,
-				 pih->pih_cont_hdl) == 0)
-			return 0;
-		ds_cont_tgt_close(pool->sp_uuid, pool->sp_srv_cont_hdl);
-		D_DEBUG(DB_MD, "delete hdl "DF_UUID"n", DP_UUID(pool->sp_srv_cont_hdl));
-		uuid_clear(pool->sp_srv_cont_hdl);
-		uuid_clear(pool->sp_srv_pool_hdl);
+	if (!uuid_is_null(pool->sp_srv_cont_hdl) &&
+	    uuid_compare(pool->sp_srv_cont_hdl, pih->pih_cont_hdl) != 0) {
+		/*
+		 * The potential handle change can only happen on the old pool (before 2.8),
+		 * following open call could be removed once 2.6 pool support is dropped
+		 */
+		D_WARN(DF_UUID ": Server cont hdl changed from " DF_UUID " to " DF_UUID "\n",
+		       DP_UUID(pool->sp_uuid), DP_UUID(pool->sp_srv_cont_hdl),
+		       DP_UUID(pih->pih_cont_hdl));
+
+		if (!uuid_is_null(pih->pih_cont_hdl))
+			rc = ds_pool_srv_open(pool->sp_uuid, pih->pih_cont_hdl);
```

Comment:
Do we need to close the old `sp_srv_cont_hdl` before open `pih_cont_hdl`?
If `ds_pool_srv_open` got failure, still need to subsequent `uuid_copy`?

Reply:
Server handle should only be closed when pool is destroyed or engine shutdown. 
Even if ds_pool_srv_open() failed, we should still propagate the global handles by following uuid_copy().

Reply:
I mean that the `pool->sp_srv_cont_hdl` has already been opened, do we need to close it before opening `pih->pih_cont_hdl`, otherwise it is overwritten?

Reply:
Yes, server handle open is just for propagating the UUID to each xstream.

Reply:
I mean that at L795, the `pih->pih_cont_hdl` does not match `pool->sp_srv_cont_hdl`, then we open the server handle with `pih->pih_cont_hdl`, and copy it into `pool->sp_srv_cont_hdl`. But before the copy, the old `pool->sp_srv_cont_hdl` may has been opened, and related `hdl` has already been into `pool_child->spc_srv_cont_hdl`, is it possible? If yes, should we close it firstly?

Reply:
no close required. "open srv handle" is just about copy the UUID somewhere, see above ds_cont_srv_open().

Reply:
Oh, right, thanks.